test/integration_test/basic_dmthin_test.go

```diff
@@ -131,6 +134,38 @@ func generateTestCases(opt OptionsArr, idx uint) map[string]types.TestCase {
 	return retVal
 }
 
+var testDmthinCases = []types.TestCase{
+	{
+		TestName:        "BasicUpgradeStorageClusterDmthinWithAllComponents",
+		TestrailCaseIDs: []string{"C50241"},
+		TestSpec: func(t *testing.T) interface{} {
+			objects, err := ci_utils.ParseSpecs("storagecluster/storagecluster-with-all-components.yaml")
+			provider := cloud_provider.GetCloudProvider()
+			require.NoError(t, err)
+			cluster, ok := objects[0].(*corev1.StorageCluster)
+			require.True(t, ok)
+			cluster.Name = "test-stc"
+			cluster.Namespace = "kube-system"
+			cloudSpec := &corev1.CloudStorageSpec{}
+			cloudSpec.DeviceSpecs = provider.GetDefaultDataDrives()
+			cloudSpec.SystemMdDeviceSpec = provider.GetDefaultMetadataDrive()
+			cluster.Spec.CloudStorage = cloudSpec
+			cluster.Spec.Monitoring.Prometheus.AlertManager.Enabled = false
+			return cluster
+		},
+		ShouldSkip: func(tc *types.TestCase) bool {
+			if len(ci_utils.PxUpgradeHopsURLList) == 0 {
+				logrus.Info("--px-upgrade-hops-url-list is empty, cannot run BasicUpgradeStorageClusterDmthinWithAllComponents test")
+				return true
+			}
+			k8sVersion, _ := k8sutil.GetVersion()
+			pxVersion := ci_utils.GetPxVersionFromSpecGenURL(ci_utils.PxUpgradeHopsURLList[0])
+			return k8sVersion.GreaterThanOrEqual(k8sutil.K8sVer1_22) && pxVersion.LessThan(pxVer2_9)
```

Comment:
Nit... DMthin is supported from 3.0.0 onwards. Let's update the pxVersion check.